src/ORM/Connect/DBSchemaManager.php

```diff
@@ -800,13 +808,35 @@ public function requireField($table, $field, $spec)
                 }
             }
             $this->transAlterField($table, $field, $spec_orig);
+            if ($this->needRebuildColumn($fieldValue, $spec_orig)) {
+                if (!isset($this->schemaUpdateTransaction[$table]['advancedOptions'])) {
+                    $this->schemaUpdateTransaction[$table]['advancedOptions'] = [];
+                }
+                $this->schemaUpdateTransaction[$table]['advancedOptions'] = array_merge(
+                    $this->schemaUpdateTransaction[$table]['advancedOptions'],
+                    ['rebuildCols' => [$field => true]],
+                );
+            }
```

Comment:
MySQL has strict rules about not allowing an `ALTER COLUMN` for generated columns in various scenarios. See `MySQLSchemaManager::needRebuildColumn()` for more details.